nibabel/gifti/gifti.py

```diff
@@ -299,26 +300,62 @@ def from_array(klass,
         return cda
 
-    def to_xml(self):
+    def _to_xml_element(self):
         # fix endianness to machine endianness
         self.endian = gifti_endian_codes.code[sys.byteorder]
-        result = ""
-        result += self.to_xml_open()
-        # write metadata
-        if not self.meta is None:
-            result += self.meta.to_xml()
-        # write coord sys
-        if not self.coordsys is None:
-            result += self.coordsys.to_xml()
+
+        data_array = xml.Element('DataArray', attrib={
+            'Intent': intent_codes.niistring[self.intent],
+            'DataType': data_type_codes.niistring[self.datatype],
+            'ArrayIndexingOrder': array_index_order_codes.label[self.ind_ord],
+            'Dimensionality': str(self.num_dim),
+            'Encoding': gifti_encoding_codes.specs[self.encoding],
+            'Endian': gifti_endian_codes.specs[self.endian],
+            'ExternalFileName': self.ext_fname,
+            'ExternalFileOffset': self.ext_offset})
+        for di, dn in enumerate(self.dims):
+            data_array.attrib['Dim%d' % di] = str(dn)
+
+        if self.meta is not None:
+            data_array.append(self.meta._to_xml_element())
+        if self.coordsys is not None:
+            data_array.append(self.coordsys._to_xml_element())
         # write data array depending on the encoding
         dt_kind = data_type_codes.dtype[self.datatype].kind
-        result += data_tag(self.data,
-                           gifti_encoding_codes.specs[self.encoding],
-                           KIND2FMT[dt_kind],
-                           self.ind_ord)
-        result = result + self.to_xml_close()
-        return result
+        data_array.append(
+            _data_tag_element(self.data,
+                              gifti_encoding_codes.specs[self.encoding],
+                              KIND2FMT[dt_kind],
+                              self.ind_ord))
+
+        return data_array
+
+    def print_summary(self):
+        print('Intent: ', intent_codes.niistring[self.intent])
+        print('DataType: ', data_type_codes.niistring[self.datatype])
+        print('ArrayIndexingOrder: ',
+              array_index_order_codes.label[self.ind_ord])
+        print('Dimensionality: ', self.num_dim)
+        print('Dimensions: ', self.dims)
+        print('Encoding: ', gifti_encoding_codes.specs[self.encoding])
+        print('Endian: ', gifti_endian_codes.specs[self.endian])
+        print('ExternalFileName: ', self.ext_fname)
+        print('ExternalFileOffset: ', self.ext_offset)
+        if not self.coordsys is None:
+            print('----')
+            print('Coordinate System:')
+            print(self.coordsys.print_summary())
+
+    @np.deprecate_with_doc("Use the metadata property instead.")
+    def get_metadata(self):
+        return self.meta.metadata
+
+    @property
+    def metadata(self):
+        """ Returns metadata as dictionary """
+        return self.meta.metadata
```

Comment:
Is this reordering intentional? PR would be a little simpler if `print_summary` and `{get_,}metadata` went back below `to_xml_*`.